packages/thirdweb/src/wallets/manager/index.ts

```diff
@@ -83,7 +83,13 @@ export function createConnectionManager(storage: AsyncStorage) {
   const isAutoConnecting = createStore(false);
 
   const connectedWallets = computedStore(() => {
-    return Array.from(walletIdToConnectedWalletMap.getValue().values());
+    const allWallets = Array.from(
+      walletIdToConnectedWalletMap.getValue().values(),
+    );
+    const hasSmartWallet = allWallets.some((w) => isSmartWallet(w));
+    return hasSmartWallet
+      ? allWallets.filter((w) => w.id !== "inApp")
```

Comment:
dont think you can do this. when doing inAppWallet with smart wallet config inside, it will only connect inApp, but isSmartWallet is also true.

instead we should handle it on connection, we already have a path that checks for 'accountAbstraction' prop. if that's the case, and the admin is an inapp wallet, only connect the smart wallet